guides/workflow/testing.md

```diff
@@ -0,0 +1,186 @@
+# Testing
+
+Testing is essential for creating and keeping up functional Wix CLI code. With Wix CLI, you can easily use popular tools like Jest and Vitest for component tests.
+
+These testing frameworks let you define expected behaviors for your code in different scenarios and then compare them with the actual outcomes.
+
+In the sections below, we'll show you how to set up specific testing frameworks with Wix CLI.
+
+## Vitest
+
+Vitest is a blazing fast unit test framework powered by Vite ([Learn more about Vitest](https://vitest.dev/guide/why.html)).
+
+Following Vitest's [getting started guide](https://vitest.dev/guide/), you can start by installing Vitest:
+
+```
+yarn install -D vitest
+```
+
+Also, install `jsdom` which we'll need for simulating a browser-like environment:
+
+```
+yarn install -D jsdom
+```
+
+Then, create the following `vitest.config.mjs` file at the root of your application:
+
+```javascript
+import { defineConfig } from 'vitest/config'
+
+export default defineConfig({
+  test: {
+    environment: 'jsdom',
+  },
+});
+```
+
+Next, add the following script to your `package.json`:
+
+```diff
+{
+    "scripts": {
++     "test:component": "vitest"
+    }
+}
+```
+
+We recommend you use [Testing Library](https://testing-library.com/) for testing your UI components. To set it up, start by installing it:
+
+```
+yarn install -D @testing-library/jest-dom @testing-library/react
+```
+
+Then, add the following to your `vitest.config.mjs`:
+
+```diff
+import { defineConfig } from 'vitest/config'
+
+export default defineConfig({
+  test: {
+    environment: 'jsdom',
++   setupFiles: ['./src/test-setup.ts'],
+  },
+});
+```
+
+Finally, create this file which sets up [Testing Library](https://testing-library.com/):
+
+```typescript
+import { expect} from 'vitest';
+import matchers from '@testing-library/jest-dom/matchers';
+import '@testing-library/jest-dom/vitest';
+
+expect.extend(matchers);
+```
```

Comment:
We've discussed this and decided to keep the test-setup for now, though this can change in the future for an import inside each test, similarly to the example in https://testing-library.com/docs/react-testing-library/example-intro.